### PR TITLE
Add a confirmation pop up when pay bounty button is clicked

### PR DESCRIPTION
**Pull request by chiragDork**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

https://www.loom.com/share/57fee31ec92a4c15b80d08b8bf325926?sid=7d5af493-53bd-4c77-9115-1edd4542171a

### CODE REVIEW

**Comment by chiragDork** [comment]
Hi, the PR has been raised, please review.

**Comment by ecurrencyhodler** [comment]
So Chiragdork is having issues with the 2nd test which is clicking "yes" and paying the bounty. He wrote the test but is unsure why it's failing. @elraphty can you take a look?

**Comment by kevkevinpal** [comment]
I see this in the stack trace
```
Cannot find module '../../../../components/common' from 'src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx'

      469 |     const mockMakePayment = jest.fn();
      470 |
    > 471 |     jest.mock('../../../../components/common', () => ({
          |          ^
      472 |       usePaymentConfirmationModal: () => ({
      473 |         openPaymentConfirmation: mockOpenPaymentConfirmation
      474 |       })

      at Resolver.resolveModule (node_modules/jest-resolve/build/resolver.js:324:11)
      at src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:471:10
      at step (src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:67:23)
      at Object.next (src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:48:53)
      at src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:42:71
      at Object.<anonymous>.__awaiter (src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:38:12)
      at Object.<anonymous> (src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:467:87)
```

**Comment by kevkevinpal** [comment]
I also see this

```
 458 |     const { getByText, queryByText } = render(<MobileView {...props} />);
      459 |
    > 460 |     const payBountyButton = getByText('Pay Bounty');
          |                             ^
      461 |     fireEvent.click(payBountyButton);
      462 |
      463 |     expect(getByText('Are you sure you want to')).toBeInTheDocument();

      at Object.getElementError (node_modules/@testing-library/dom/dist/config.js:37:19)
      at apply (node_modules/@testing-library/dom/dist/query-helpers.js:76:38)
      at apply (node_modules/@testing-library/dom/dist/query-helpers.js:52:25)
      at node_modules/@testing-library/dom/dist/query-helpers.js:95:24
      at Object.<anonymous> (src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:460:29)
```

File: src/components/common/PaymentConfirmationModal/PaymentConfirmationModal.tsx
Comment (kevkevinpal) [inline src/components/common/PaymentConfirmationModal/PaymentConfirmationModal.tsx:1]: For this file lets make a new component test in `src/components/common/PaymentConfirmationModal/__tests__/PaymentConfirmationModal.spec.tsx`

File: src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx [hunk 1]
```diff
@@ -446,4 +446,57 @@ describe('MobileView component', () => {
...
+
+    expect(getByText('Are you sure you want to')).toBeInTheDocument();
+    expect(queryByText('Pay this Bounty?')).toBeInTheDocument();
+  });
+
+  it('marks the bounty as paid when the confirmation screen "Yes" button is clicked', async () => {
+    const mockOpenPaymentConfirmation = jest.fn();
+    const mockMakePayment = jest.fn();
+
+    jest.mock('../../../../components/common', () => ({
```
Comment (kevkevinpal) [inline src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:471]: I think this path may be incorrect
should be
`../../../../../components/common`

File: src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx [hunk 2]
```diff
@@ -379,6 +380,23 @@ function MobileView(props: CodingBountiesProps) {
   const hasAccess = isOwner || userBountyRole;
   const payBountyDisable = !isOwner && !userBountyRole;
 
+  const { openPaymentConfirmation } = usePaymentConfirmationModal();
+
+  const confirmPaymentHandler = () => {
+    console.log('here');
```
Comment (kevkevinpal) [inline src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:386]: delete this

**Review by kevkevinpal — COMMENTED** [review]
the code seems to be fine just needs some modifications on the tests otherwise good job!

**Comment by chiragDork** [comment]
Hi @kevkevinpal I made the mentioned changes can you please check. Thanks.

**Comment by ecurrencyhodler** [comment]
Tested and paid!

**Pull request merged**